Objects/listobject.c

```diff
@@ -1320,6 +1320,35 @@ list_extend_dict(PyListObject *self, PyDictObject *dict, int which_item)
     return 0;
 }
 
+static int
+list_extend_dictitems(PyListObject *self, PyDictObject *dict)
+{
+    Py_ssize_t m = Py_SIZE(self);
+    Py_ssize_t n = PyDict_GET_SIZE(dict);
+    if (list_resize(self, m + n) < 0) {
+        return -1;
+    }
+
+    // TODO: Don't start a garbage collection cycle when creating tuples
+    PyObject **dest = self->ob_item + m;
+    Py_ssize_t pos = 0;
+    Py_ssize_t i = 0;
+    PyObject *key, *value;
+    while (_PyDict_Next((PyObject *)dict, &pos, &key, &value, NULL)) {
+        PyObject *item = PyTuple_Pack(2, key, value);
+        if (item == NULL) {
+            Py_SET_SIZE(self, m + i);
+            return -1;
+        }
+        FT_ATOMIC_STORE_PTR_RELAXED(*dest, item);
```

Comment:
I think we want `RELEASE` here.

We probably want to do the same thing for the other `list_extend` functions.